platform/linux-generic/odp_packet.c

```diff
@@ -2950,3 +2950,154 @@ odp_packet_reass_partial_state(odp_packet_t pkt, odp_packet_t frags[],
 	(void)res;
 	return -ENOTSUP;
 }
+
+void *odp_packet_buf_head(odp_packet_buf_t pkt_buf)
+{
+	odp_packet_hdr_t *pkt_hdr = (odp_packet_hdr_t *)(uintptr_t)pkt_buf;
+	pool_t *pool = pkt_hdr->buf_hdr.pool_ptr;
+	uint32_t head_offset = sizeof(odp_packet_hdr_t) + pool->ext_param.pkt.app_header_size;
+
+	if (pool->pool_ext == 0) {
+		ODP_ERR("Not an external memory pool\n");
+		return NULL;
+	}
+
+	return (uint8_t *)pkt_hdr + head_offset;
+}
+
+uint32_t odp_packet_buf_size(odp_packet_buf_t pkt_buf)
+{
+	odp_packet_hdr_t *pkt_hdr = (odp_packet_hdr_t *)(uintptr_t)pkt_buf;
+	pool_t *pool = pkt_hdr->buf_hdr.pool_ptr;
+	uint32_t head_offset = sizeof(odp_packet_hdr_t) + pool->ext_param.pkt.app_header_size;
+
+	return pool->ext_param.pkt.buf_size - head_offset;
+}
+
+uint32_t odp_packet_buf_data_offset(odp_packet_buf_t pkt_buf)
+{
+	odp_packet_hdr_t *pkt_hdr = (odp_packet_hdr_t *)(uintptr_t)pkt_buf;
+
+	return (uintptr_t)pkt_hdr->seg_data - (uintptr_t)odp_packet_buf_head(pkt_buf);
+}
+
+uint32_t odp_packet_buf_data_len(odp_packet_buf_t pkt_buf)
+{
+	odp_packet_hdr_t *pkt_hdr = (odp_packet_hdr_t *)(uintptr_t)pkt_buf;
+
+	return pkt_hdr->seg_len;
+}
```

Comment:
Some of these smaller functions would be rather easy to inline but this can be done in a separate PR.